driver/main.ml

```diff
@@ -125,6 +125,7 @@ module Options = Main_args.Make_bytecomp_options (struct
   let _w = (Warnings.parse_options false)
   let _warn_error = (Warnings.parse_options true)
   let _warn_help = Warnings.help_warnings
+  let _color = Location.enable_color
```

Comment:
I am unconvinced that this is the right interface for this code. It would be more standard to have a `Cflags.color : bool ref`. Doing this would force you to have some logic to initialize color-printing (a `Location.initialize_color` function doing much of what this one does today) on the first location to print; but that would not increase the code complexity much as you already have this only-on-the-first-call logic anyway.

Also something global outside `Location` scales better to a future where other aspects of the compiler output could be colored (`-help`, `-warn-help` for example).